# Operators, If-Else, Modules & Loops


## Operators in Python

Operators are **special symbols** that perform operations on values (operands).



### Types of Operators

| **Category**   | **Operators**                          | **Description**            |
|----------------|----------------------------------------|----------------------------|
| **Arithmetic** | `+`, `-`, `*`, `/`, `//`, `%`, `**`    | Mathematical operations    |
| **Relational** | `==`, `!=`, `>`, `<`, `>=`, `<=`       | Comparison between values  |
| **Logical**    | `and`, `or`, `not`                     | Boolean logic              |
| **Bitwise**    | `&`, `^`, `~`, `<<`, `>>`         | Binary-level operations    |
| **Assignment** | `=`, `+=`, `-=`, `*=`, `/=`, `%=`, `**=`, `//=` | Assign/update values |
| **Membership** | `in`, `not in`                         | Check existence in sequence|




#### Arithmetic Operators

In [1]:
# Arithmetic Operators

print(5+6)    # Addition:        11

print(5-6)    # Subtraction:     -1

print(5*6)    # Multiplication:  30

print(5/2)    # True Division:   2.5

print(5//2)   # Floor Division:  2

print(5%2)    # Modulus:         1

print(5**2)   # Exponentiation:  25

11
-1
30
2.5
2
1
25




| Operator | Name | Example | Result |
|----------|------|---------|--------|
| `+` | Addition | `5+6` | `11` |
| `-` | Subtraction | `5-6` | `-1` |
| `*` | Multiplication | `5*6` | `30` |
| `/` | True Division | `5/2` | `2.5` (always float) |
| `//` | Floor Division | `5//2` | `2` (rounds down) |
| `%` | Modulus (remainder) | `5%2` | `1` |
| `**` | Exponentiation | `5**2` | `25` |

#### Relational (Comparison) Operators

In [2]:
# Relational Operators

print(4>5)    # Greater than:          False

print(4<5)    # Less than:             True

print(4>=4)   # Greater than or equal: True

print(4<=4)   # Less than or equal:    True

print(4==4)   # Equal to:              True

print(4!=4)   # Not equal to:          False

False
True
True
True
True
False



All relational operators return a **Boolean** (`True` or `False`).

#### Logical Operators

In [3]:
# Logical Operators

print(1 and 0)    # AND: both must be True -> 0 (False)

print(1 or 0)     # OR: at least one True -> 1 (True)

print(not 1)      # NOT: reverses the value -> False

0
1
False



| Operator | Description | Example | Result |
|----------|-------------|---------|--------|
| `and` | True if both are True | `1 and 0` | `0` (False) |
| `or` | True if at least one is True | `1 or 0` | `1` (True) |
| `not` | Reverses the value | `not 1` | `False` |

**Note:** In Python, `0` is False and any non-zero number is True.

#### Bitwise Operators

In [4]:
# Bitwise Operators

# bitwise and
print(2 & 3)    # AND: 10 & 11 = 10 = 2

# bitwise or
print(2 | 3)    # OR:  10 | 11 = 11 = 3

# bitwise xor
print(2 ^ 3)    # XOR: 10 ^ 11 = 01 = 1

print(~3)       # NOT: -(3+1) = -4

print(4 >> 2)   # Right Shift: 4 / 2^2 = 1

print(5 << 2)   # Left Shift:  5 * 2^2 = 20

2
3
1
-4
1
20



| Operator | Description | Example | Binary | Result |
|----------|-------------|---------|--------|--------|
| `&` | AND | `2 & 3` | `10 & 11 = 10` | `2` |
| `|` | OR | `2 | 3` | `10 | 11 = 11` | `3` |
| `^` | XOR | `2 ^ 3` | `10 ^ 11 = 01` | `1` |
| `~` | NOT | `~3` | `-(3+1)` | `-4` |
| `>>` | Right Shift | `4 >> 2` | `4 / 2^2` | `1` |
| `<<` | Left Shift | `5 << 2` | `5 * 2^2` | `20` |

#### Assignment Operators

In [5]:
# Assignment Operators

a = 2

a %= 2    # a = a % 2 -> 0

print(a)

0



| Operator | Equivalent |
|----------|------------|
| `=` | `a = b` |
| `+=` | `a = a + b` |
| `-=` | `a = a - b` |
| `*=` | `a = a * b` |
| `/=` | `a = a / b` |
| `//=` | `a = a // b` |
| `%=` | `a = a % b` |
| `**=` | `a = a ** b` |


#### Membership Operators

In [6]:
# Membership Operators (in/not in)

print('D' not in 'Delhi')    # False ('D' IS in 'Delhi')

print(1 in [2,3,4,5,6])      # False (1 is NOT in the list)

False
False



| Operator | Description | Example | Result |
|----------|-------------|---------|--------|
| `in` | True if value is in sequence | `'D' in 'Delhi'` | `True` |
| `not in` | True if value is not in sequence | `'D' not in 'Delhi'` | `False` |

### Program: Sum of Digits of a 3-Digit Number

In [8]:
# Program - Find the sum of a 3 digit number entered by the user
number = int(input('Enter a 3 digit number'))

# 345%10 -> 5
a = number%10
number = number//10
# 34%10 -> 4
b = number % 10
number = number//10
# 3 % 10 -> 3
c = number % 10

print(a + b + c)

Enter a 3 digit number 123


6




**Explanation:**
- `666 % 10 = 6` (last digit)
- `666 // 10 = 66`
- `66 % 10 = 6` (middle digit)
- `66 // 10 = 6`
- `6 % 10 = 6` (first digit)
- Sum = `6 + 6 + 6 = 18`

# If-Else in Python

Conditional statements let your program make decisions.

### Syntax:
```python
if condition:
    # runs if True
elif another_condition:
    # runs if this is True
else:
    # runs if all are False
```

> **Important:** Python uses **indentation** (4 spaces) instead of `{}` braces.

### Login Program with Indentation

In [9]:
# login program and indentation
# email -> nitish.campusx@gmail.com
# password -> 1234
email = input('enter email')
password = input('enter password')

if email == 'nitish.campusx@gmail.com' and password == '1234':
    print('Welcome')
elif email == 'nitish.campusx@gmail.com' and password != '1234':
    # tell the user
    print('Incorrect password')
    password = input('enter password again')
    if password == '1234':
        print('Welcome,finally!')
    else:
        print('beta tumse na ho paayega!')
else:
    print('Not correct')

enter email 1986ravnoorsingh8the@gmail.com
enter password 12422


Not correct





**Explanation:**
- If email is wrong -> `else` block runs
- If email is right but password is wrong -> `elif` block runs, asks for password again
- If both correct -> `if` block runs

### Min of 3 Numbers

In [10]:
# min of 3 numbers
a = int(input('first num'))
b = int(input('second num'))
c = int(input('third num'))

if a<b and a<c:
    print('smallest is',a)
elif b<c:
    print('smallest is',b)
else:
    print('smallest is',c)

first num 123
second num 45
third num 235


smallest is 45


### Menu Driven Calculator

In [11]:
# menu driven calculator
menu = input("""
Hi! how can I help you.
1. Enter 1 for pin change
2. Enter 2 for balance check
3. Enter 3 for withdrawl
4. Enter 4 for exit
""")

if menu == '1':
    print('pin change')
elif menu == '2':
    print('balance')
else:
    print('exit')


Hi! how can I help you.
1. Enter 1 for pin change
2. Enter 2 for balance check
3. Enter 3 for withdrawl
4. Enter 4 for exit
 1


pin change


# Modules in Python

Modules are **pre-built libraries** that provide extra functionality. Import them using `import`.

### Common Modules:

| Module | Purpose |
|--------|---------|
| `math` | Mathematical functions |
| `keyword` | Python keywords list |
| `random` | Random number generation |
| `datetime` | Date and time functions |

In [12]:
# math module
import math

print(math.sqrt(196))  # Square root

print(math.pi)         # Value of pi

print(math.factorial(5))  # 5! = 120

14.0
3.141592653589793
120


In [13]:
# keyword module
import keyword
print(keyword.kwlist)

['False', 'None', 'True', 'and', 'as', 'assert', 'async', 'await', 'break', 'class', 'continue', 'def', 'del', 'elif', 'else', 'except', 'finally', 'for', 'from', 'global', 'if', 'import', 'in', 'is', 'lambda', 'nonlocal', 'not', 'or', 'pass', 'raise', 'return', 'try', 'while', 'with', 'yield']


In [14]:
# random module
import random
print(random.randint(1,100))  # Random integer between 1 and 100

90


In [15]:
# datetime module
import datetime
print(datetime.datetime.now())

2026-08-17 22:32:49.246861


In [16]:
help('modules')


Please wait a moment while I gather a list of all available modules...

test_sqlite3: testing with SQLite version 3.50.4
81d243bd2c585b0f4821__mypyc abc                 iniconfig           queue
IPython             altair              inspect             quopri
PIL                 annotated_doc       io                  random
__future__          annotated_types     ipaddress           re
__hello__           annotationlib       ipykernel           referencing
__phello__          antigravity         ipykernel_launcher  reprlib
_abc                anyio               ipython_pygments_lexers requests
_aix_support        argon2              isoduration         rfc3339_validator
_android_support    argparse            itertools           rfc3986_validator
_apple_support      array               itsdangerous        rfc3987_syntax
_argon2_cffi_bindings arrow               jedi                rlcompleter
_ast                ast                 jinja2              rpds
_ast_unparse        astt

# Loops in Python

Loops let you **repeat code** multiple times.

### Types:
- **while loop** - Repeats while a condition is True
- **for loop** - Iterates over a sequence

### While Loop

In [17]:
# While loop - Print multiplication table
number = int(input('enter the number'))
i = 1

while i < 11:
    print(number,'*',i,'=',number * i)
    i += 1

enter the number 34


34 * 1 = 34
34 * 2 = 68
34 * 3 = 102
34 * 4 = 136
34 * 5 = 170
34 * 6 = 204
34 * 7 = 238
34 * 8 = 272
34 * 9 = 306
34 * 10 = 340


### While Loop with Else

In [18]:
# while loop with else
x = 1
while x < 3:
    print(x)
    x += 1
else:
    print('limit crossed')

1
2
limit crossed



The `else` block runs when the while condition becomes False (without `break`).

### Guessing Game

In [19]:
# Guessing game
import random
jackpot = random.randint(1,100)
guess = int(input('guess karo'))
counter = 1

while guess != jackpot:
    if guess < jackpot:
        print('galat! guess higher')
    else:
        print('galat! guess lower')
    guess = int(input('guess karo'))
    counter += 1
else:
    print('correct guess')
    print('attempts',counter)

guess karo 45


galat! guess higher


guess karo 6


galat! guess higher


guess karo 100


galat! guess lower


guess karo 50


galat! guess higher


guess karo 60


galat! guess higher


guess karo 80


galat! guess higher


guess karo 90


galat! guess higher


guess karo 95


galat! guess higher


guess karo 97


galat! guess higher


guess karo 98


galat! guess higher


guess karo 99


correct guess
attempts 11


### For Loop

In [20]:
# For loop demo
for i in {1,2,3,4,5}:
    print(i)

1
2
3
4
5


### For Loop Examples

#### Population Growth Problem
The current population of a town is 10000. The population is increasing at 10% per year. Find the population at the end of each of the last 10 years.

In [21]:
# Population growth (working backwards)
curr_pop = 10000
for i in range(10,0,-1):
    print(i,curr_pop)
    curr_pop = curr_pop - 0.1*curr_pop

10 10000
9 9000.0
8 8100.0
7 7290.0
6 6561.0
5 5904.9
4 5314.41
3 4782.969
2 4304.6721
1 3874.20489



**Explanation:** Each year we subtract 10% of the current population. `range(10,0,-1)` counts down from 10 to 1.

In [23]:
#Correct Answer of abovequestion:
curr_pop = 10000
for i in range(10,0,-1):
    print(i,curr_pop)
    curr_pop/= 1.1

10 10000
9 9090.90909090909
8 8264.462809917353
7 7513.148009015775
6 6830.134553650703
5 6209.213230591548
4 5644.739300537771
3 5131.5811823070635
2 4665.07380209733
1 4240.976183724845


# Nested Loops

A loop inside another loop.

### Pattern 1: Triangle

In [24]:
# Pattern 1
for i in range(1,5):
    for j in range(i):
        print('*',end=' ')
    print()

* 
* * 
* * * 
* * * * 


### Pattern 2: Number Pyramid

In [25]:
# Pattern 2
for i in range(1,5):
    for j in range(1,i+1):
        print(j,end=' ')
    print()

1 
1 2 
1 2 3 
1 2 3 4 


# Loop Control Statements

| Statement | Description |
|-----------|-------------|
| `break` | Exits the loop immediately |
| `continue` | Skips the current iteration, goes to next |
| `pass` | Does nothing (placeholder) |

### Break Statement

In [26]:
# Break demo
for i in range(1,11):
    if i == 5:
        break
    print(i)

1
2
3
4


### Continue Statement

In [27]:
# Continue demo - skip 5
for i in range(1,11):
    if i == 5:
        continue
    print(i)

1
2
3
4
6
7
8
9
10


### Pass Statement

In [28]:
# Pass demo - does nothing, just a placeholder
for i in range(1,6):
    if i == 3:
        pass  # will implement later
    print(i)

1
2
3
4
5


````markdown
#### Population Growth Calculation

To calculate the population for each previous year when there is a **10% annual increase**, we can use a simple equation based on the population of the previous year.

Let the population of the previous year be represented by **x**.

##### 1. Calculate the Current Year's Population

If the population increases by **10%** every year:

```text
Current Year Population = x × 1.1
````

Here:

* `x` = Population of the previous year
* `1.1` = 100% + 10% increase

For example, if the previous year's population is `10,000`:

```text
Current Year Population = 10,000 × 1.1
                        = 11,000
```

##### 2. Calculate the Previous Year's Population

If we already know the current year's population and want to find the population of the previous year, rearrange the equation:

```text
Current Year Population = x × 1.1
```

Divide both sides by `1.1`:

```text
x = Current Year Population / 1.1
```

Therefore:

```text
Previous Year Population = Current Year Population / 1.1
```

##### Example

Suppose the current year's population is:

```text
10,000
```

Then the previous year's population would be:

```text
Previous Year Population = 10,000 / 1.1
                         ≈ 9,090.91
```

##### 3. Finding Population for Multiple Previous Years

The same calculation can be repeated for every previous year:

```text
Previous Year 1 = Current Population / 1.1

Previous Year 2 = Previous Year 1 / 1.1

Previous Year 3 = Previous Year 2 / 1.1
```

In general:

```text
Previous Year Population = Current Year Population / 1.1
```

This method allows us to calculate the population for each previous year when the population is assumed to have increased by **10% every year**.

```
```


# Practice Problems

15 problems covering operators, if-else, loops, and modules.

### Problem 1: Monthly Salary After Deductions

Write a program that will give you in-hand monthly salary after deduction on CTC - HRA(10%), DA(5%), PF(3%) and taxes deduction as below:

| Salary (Lakhs) | Tax (%) |
|----------------|----------|
| Below 5 | 0% |
| 5-10 | 10% |
| 10-20 | 20% |
| Above 20 | 30% |

In [30]:
# Write code here
ctc = int(input('Enter your anual CTC:'))

if ctc < 500000:
    salary=ctc*.82
elif ctc < 1000000:
    salary=ctc*.72
elif ctc < 2000000:
    salary=ctc*.62
else:
    salary=ctc*.52

print("You in hand monthly salary will be-", round(salary/12,2))

Enter your anual CTC: 10000000


You in hand monthly salary will be- 433333.33




**Explanation:** 12 lakhs falls in 10-20 bracket (20% tax). After HRA+DA+PF deductions (18%) and tax (20%), remaining = 62%. Monthly = `1200000 * 0.62 / 12` = `62000`

### Problem 2: Triangle Validation

Write a program that takes three angles and checks if they can form a triangle.

**Hint:** Sum of all angles must be 180 and all angles must be positive.

In [31]:
# Write code here
first = int(input('enter the 1st angle'))
second = int(input('enter the 2nd angle'))
third = int(input('enter the 3rd angle'))

if (first+second+third) == 180 and first>0 and second>0 and third>0:
    print('forms a triangle')
else:
    print('does not form a triangle')

enter the 1st angle 45
enter the 2nd angle 55
enter the 3rd angle 100


does not form a triangle


### Problem 3: Profit or Loss

Write a program that takes cost price and selling price and determines whether its a loss or a profit.

In [32]:
# Write code here
cost_price = int(input('Enter cost price-'))
selling_price = int(input('Enter selling price-'))

if cost_price < selling_price:
    print('Profit')
elif cost_price > selling_price:
    print('Loss')
else:
    print('No Loss No Gain')

Enter cost price- 100
Enter selling price- 190


Profit


### Problem 4: Menu-Driven Unit Converter

Write a menu-driven program for:
- cm to ft (1 cm = 0.032 ft)
- km to miles (1 km = 0.62 miles)
- USD to INR (1 USD = 80 INR)
- exit

In [33]:
# Write code here
menu = input("""
Hi select an option
1. cms to ft
2. km to miles
3. USD to INR
4. Exit
""")

if menu == '1':
    cm = float(input('enter the cm value'))
    print('ft value is',0.032*cm)
elif menu == '2':
    km = float(input('enter the km value'))
    print('miles value is',km*0.62)
elif menu == '3':
    usd = float(input('enter usd'))
    print('inr',usd*80)
else:
    exit()


Hi select an option
1. cms to ft
2. km to miles
3. USD to INR
4. Exit
 1
enter the cm value 34


ft value is 1.088


### Problem 5: Fibonacci Series (10 terms)

Display Fibonacci series up to 10 terms.

**Fibonacci Sequence:** 0, 1, 1, 2, 3, 5, 8, 13, 21, 34...

Each number is the sum of the two before it.

In [34]:
# Write code here
num1,num2 = 0,1

for i in range(10):
    print(num1)
    next = num1 + num2
    num1 = num2
    num2 = next

0
1
1
2
3
5
8
13
21
34


### Problem 6: Factorial of a Number

Find the factorial of a given number.

**5!** = 5 x 4 x 3 x 2 x 1 = **120**

In [36]:
# Write code here
num = int(input('enter the number'))

fact = 1
for i in range(1,num+1):
    fact = fact*i

print(fact)

enter the number 5


120


### Problem 7: Reverse a Number

Reverse a given integer number.

**Input:** 76542 -> **Output:** 24567

In [37]:
# Write code here
number = int(input('enter the number'))

rev = 0
while number>0:
    last = number%10
    rev = rev*10 + last
    number = number//10

print(rev)

enter the number 12334


43321


`

**Logic:** Extract last digit with `%10`, add to reversed number, remove last digit with `//10`.

### Problem 8: Sum 1 to N (Skip Divisible by 5, Stop at 300)

Find sum from 1 to N. Skip numbers divisible by 5. Stop if sum exceeds 300. Don't use for loop.

In [38]:
# Write code here
N = int(input('enter the number'))
sum = 0
i = 1

while i < N+1:
    if i % 5 == 0:
        i+=1
        continue

    sum += i

    if sum > 300:
        sum = sum - i
        break

    i+=1

print(sum)

enter the number 50


276




**Key concepts:** `continue` to skip divisible by 5, `break` when sum > 300.

### Problem 9: Sum and Average Until Zero

Keep accepting numbers until user enters Zero. Display sum and average.

In [39]:
# Write code here
sum = 0
count = 0

while True:
    num = int(input('enter number'))
    if num == 0:
        break
    sum = sum + num
    count = count + 1

print('sum',sum)
print('avg',sum/count)

enter number 34
enter number 45
enter number 232
enter number 45
enter number 0


sum 356
avg 89.0


### Problem 10: Divisible by 7 but Not Multiple of 5

Find all numbers between 2000 and 3200 that are divisible by 7 but not a multiple of 5.

In [40]:
# Write code here
L = []
for i in range(2000,3201):
    if i % 7 == 0 and i % 5 != 0:
        L.append(str(i))

print(",".join(L))

2002,2009,2016,2023,2037,2044,2051,2058,2072,2079,2086,2093,2107,2114,2121,2128,2142,2149,2156,2163,2177,2184,2191,2198,2212,2219,2226,2233,2247,2254,2261,2268,2282,2289,2296,2303,2317,2324,2331,2338,2352,2359,2366,2373,2387,2394,2401,2408,2422,2429,2436,2443,2457,2464,2471,2478,2492,2499,2506,2513,2527,2534,2541,2548,2562,2569,2576,2583,2597,2604,2611,2618,2632,2639,2646,2653,2667,2674,2681,2688,2702,2709,2716,2723,2737,2744,2751,2758,2772,2779,2786,2793,2807,2814,2821,2828,2842,2849,2856,2863,2877,2884,2891,2898,2912,2919,2926,2933,2947,2954,2961,2968,2982,2989,2996,3003,3017,3024,3031,3038,3052,3059,3066,3073,3087,3094,3101,3108,3122,3129,3136,3143,3157,3164,3171,3178,3192,3199



**Logic:** Check `i % 7 == 0` (divisible by 7) AND `i % 5 != 0` (not multiple of 5).

### Problem 11: Robot Movement Distance

A robot moves from origin (0,0). Compute distance after sequence of movements (UP, DOWN, LEFT, RIGHT).

In [44]:
# Write code here
pos = [0,0]
while True:
    s = input('Enter the robot path')
    if s == '!':
        break
    direction = s.split()[0]
    steps = int(s.split()[1])

    if direction == 'UP':
        pos[1] = pos[1] + steps
    elif direction == 'DOWN':
        pos[1] = pos[1] - steps
    elif direction == 'LEFT':
        pos[0] = pos[0] - steps
    elif direction == 'RIGHT':
        pos[0] = pos[0] + steps

print('new pos',pos)
print((pos[0]**2 + pos[1]**2)**0.5)

Enter the robot path UP 5
Enter the robot path DOWN 3
Enter the robot path LEFT 3
Enter the robot path RIGHT 2
Enter the robot path !


new pos [-1, 2]
2.23606797749979


### Problem 12: Prime Number Check

Write a program to print whether a given number is prime or not.

In [46]:
# Write code here
num = int(input('enter the num'))

flag = True
for i in range(2,num):
    if num%i == 0:
        flag = False
        break

if flag == True:
    print('Prime')
else:
    print('Not Prime')

enter the num 29


Prime



**Logic:** If any number from 2 to num-1 divides evenly, it's not prime.

### Problem 13: Armstrong Numbers in Range

Print all Armstrong numbers in a given range.

**Armstrong number:** Sum of digits each raised to power n equals the number itself.

Example: `153 = 1^3 + 5^3 + 3^3 = 153`

In [50]:
# Write code here
start = int(input("Enter the start of the range: "))
end = int(input("Enter the end of the range: "))

for num in range(start, end+1):
    temp = num
    su_m = 0
    order = len(str(num))
    while temp > 0:
        digit = temp % 10
        su_m += digit ** order
        temp //= 10
    if num == su_m:
        print(num)

Enter the start of the range:  1
Enter the end of the range:  200


1
2
3
4
5
6
7
8
9
153



**Explanation:**
- `order = len(str(num))` gets the number of digits
- Extract each digit with `% 10`, raise to power `order`, sum them
- If sum equals original number, it's an Armstrong number

### Problem 14: Clock Angle Between Hands

Calculate the minimum angle between hour and minute hand.

**Key facts:**
- 1 hour = 30 degrees (360/12)
- 1 minute = 6 degrees (360/60)
- Hour hand moves 0.5 degrees per minute

In [51]:
# Write code here
h = int(input('Enter hours hand-'))
m = int(input('Enter minute hand-'))

# validate the input
if (h < 0 or m < 0 or h > 12 or m > 60):
    print('Wrong input')

# minute angle (1 min = 6 degrees)
m_angle = m*6

# hour angle (1 hour = 30 degrees + 0.5 per minute)
h_angle = h*30 + m*0.5

# Take abs difference
angle = abs(h_angle - m_angle)

if angle>180:
    print(360-angle)
else:
    print(angle)

Enter hours hand- 12
Enter minute hand- 34


173.0



**Explanation:**
- Hour hand at 9: `9 * 30 = 270` degrees
- Minute hand at 0: `0 * 6 = 0` degrees
- Difference: `270` degrees
- Since 270 > 180: `360 - 270 = 90` degrees

### Problem 15: Rectangle Overlap Check

Given two rectangles, find if they overlap. Each rectangle is defined by top-left (L) and bottom-right (R) coordinates.

In [53]:
# Write code here
print("Enter both Rectangle points coordinates <x y>")

l1_x, l1_y = list(map(int, input('Enter L1 coordinate value (Two value space seperated):').split()))
r1_x, r1_y = list(map(int, input('Enter r1 coordinate value,(Two value space seperated):').split()))

l2_x, l2_y = list(map(int, input('Enter L2 coordinate value,(Two value space seperated):').split()))
r2_x, r2_y = list(map(int, input('Enter r2 coordinate value,(Two value space seperated):').split()))

# if rectangle has area 0, no overlap
if l1_x == r1_x or l1_y == r1_y or r2_x == l2_x or l2_y == r2_y:
    print("Don't Overlap")

# If one rectangle is on left side of other
elif l1_x > r2_x or l2_x > r1_x:
    print("Don't Overlap")

# If one rectangle is above other
elif r1_y > l2_y or r2_y > l1_y:
    print("Don't Overlap")
else:
    print("Overlap")

Enter both Rectangle points coordinates <x y>


Enter L1 coordinate value (Two value space seperated): 12 34
Enter r1 coordinate value,(Two value space seperated): 124 45
Enter L2 coordinate value,(Two value space seperated): 13 45
Enter r2 coordinate value,(Two value space seperated):  13 45


Don't Overlap


**Logic:**
1. Check if either rectangle has zero area
2. Check if one is entirely to the left of the other
3. Check if one is entirely above the other
4. If none of the above -> they overlap

## Summary

| Topic | Key Points |
|-------|------------|
| **Operators** | Arithmetic, Relational, Logical, Bitwise, Assignment, Membership |
| **If-Else** | `if`, `elif`, `else`, indentation-based blocks |
| **Modules** | `math`, `keyword`, `random`, `datetime` |
| **While Loop** | Repeats while condition is True |
| **For Loop** | Iterates over sequences, `range()` |
| **Nested Loops** | Loop inside a loop |
| **Control Statements** | `break`, `continue`, `pass` |

